src/libs/actions/OnyxUpdateManager.js

```diff
@@ -0,0 +1,78 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Log from '../Log';
+import * as SequentialQueue from '../Network/SequentialQueue';
+import * as App from './App';
+import * as OnyxUpdates from './OnyxUpdates';
+
+// This file is in charge of looking at the updateIDs coming from the server and comparing them to the last updateID that the client has.
+// If the client is behind the server, then we need to pause everything, get the missing updates from the server, apply those updates,
+// then restart everything. This will ensure that the client is up-to-date with the server and all the updates have been applied
+// in the correct order.
+// It's important that this file is separate and not imported by OnyxUpdates.js, so that there are no circular dependencies. Onyx
+// is used as a pub/sub mechanism to break out of the circular dependencies.
+// The circular dependency happen because this file calls the API GetMissingOnyxUpdates which uses the SaveResponseInOnyx.js file
+// (as a middleware). Therefore, SaveResponseInOnyx.js can't import and use this file directly.
+
+// This key needs to be separate from ONYXKEYS.ONYX_UPDATES_FROM_SERVER so that it can be updated without triggering the callback when the server IDs are updated
```

Comment:
This comment probably isn't as necessary here as it is in the other location.

Reply:
Removed